### DISCUSSION TITLE
improve typing of SegmentedControl

### DISCUSSION THREAD

**Discussion by Jonatan-Rydberg-work-account** [mantinedev/mantine#6545] [Feature requests]
Please consider changing the signature of `SegmentedControl(props)` to 
``` typescript
SegmentedControl<T extends string>(props: {
    value: T, 
    onChange: ((value: T) => void),
    data: (T | SegmentedControlItem<T>)[],
    ...otherProps
})
```
Also change `SegmentedControlItem` to 
``` typescript
export interface SegmentedControlItem<T extends string> {
  value: T;
  label: React.ReactNode;
  disabled?: boolean;
}
```

This will help with union types similar to `type OptionType = "Option1" | "Option2"`

**Comment by rtivital** [comment]
Generic components are tedious to maintain. I consider current types to be strong enough

**Reply by sakgoyal** [reply]
is this still the case? I would really appreciate this being generic if possible
